src/Tools/NodeDocumentationMarkdownGenerator/Commands/FromDirectoryCommand.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NodeDocumentationMarkdownGenerator.Verbs;
+
+namespace NodeDocumentationMarkdownGenerator.Commands
+{
+    internal static class FromDirectoryCommand
+    {
+        /// <summary>
+        /// Creates markdown files using the fromdirectory verb
+        /// </summary>
+        /// <param name="opts"></param>
+        internal static void HandleDocumentationFromDirectory(FromDirectoryOptions opts)
+        {
+            var searchOption = opts.RecursiveScan ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly;
+            var fileInfos = new List<MdFileInfo>();
+            fileInfos.AddRange(ScanAssembliesFromOpts(opts.InputFolderPath, opts.Filter, opts.ReferencePaths, searchOption));
+            if (opts.IncludeCustomNodes)
+            {
+                fileInfos.AddRange(ScanFolderForCustomNodes(opts.InputFolderPath, searchOption));
+            }
+
+            MarkdownHandler.CreateMdFilesFromFileNames(fileInfos, opts.OutputFolderPath, opts.Overwrite, opts.CompressImages, opts.DictionaryDirectory, opts.LayoutSpecPath);
+        }
+
+        private static List<MdFileInfo> ScanFolderForCustomNodes(string inputFolderPath, SearchOption searchOption)
+        {
+            var allDyfs = Directory.GetFiles(inputFolderPath, "*.dyf", searchOption).Select(x => new FileInfo(x)).ToList();
+            var fileInfos = new List<MdFileInfo>();
+            foreach (var cn in allDyfs)
+            {
+                var fileInfo = MdFileInfo.FromCustomNode(cn.FullName);
+                if (fileInfo is null) continue;
+
+                fileInfos.Add(fileInfo);
+            }
+            return fileInfos;
+        }
+
+        private static List<MdFileInfo> ScanAssembliesFromOpts(string inputFolderPath, IEnumerable<string> filter, IEnumerable<string> references, SearchOption searchOption)
+        {
+            var allDlls = Directory.GetFiles(inputFolderPath, "*.dll", searchOption).Select(x => new FileInfo(x)).ToList();
+
+            var referencesDllPaths = references
+                .SelectMany(p => new DirectoryInfo(p)
+                    .EnumerateFiles("*.dll", SearchOption.AllDirectories)
+                    .Select(d => d.FullName)
+                    .ToList());
+
+            if (filter.Count() != 0)
+            {
+                var dllPaths = allDlls
+                    .Where(x => filter.Contains(x.Name) || filter.Contains(x.FullName))
+                    .Select(x => x.FullName)
+                    .ToList();
+
+                var addtionalPathsToLoad = allDlls.Select(x => x.FullName).Except(dllPaths).ToList();
```

Comment:
I need some help here - shouldn't this be `dllPaths`? Might help if there were some comments on what each subsection is filtering or trying to achieve.

Reply:
added some comments and cleaned this up a bit, hope its a bit more clear now